controller/scan/registry.go

```diff
@@ -1544,29 +1550,21 @@ func (rs *Registry) scheduleScanImages(
 					scanTypesRequired.Vulnerability = true
 				}
 
-				signatureInfoChanged := info.SignatureDigest != sum.SignatureDigest
-				sigstoreConfigurationChanged := sigstoreTimestamp != sum.SigstoreTimestamp
-
-				if signatureInfoChanged {
+				// determine if signature scan is required
+				if !newImage {
+					scanTypesRequired.Signature = false
+				} else if signatureInfoChanged {
+					smd.scanLog.WithFields(log.Fields{"registry": rs.config.Name, "image": image, "status": sum.Status, "imageID": info.ID}).Debug("Signature info for image changed, signature scan required.")
+					scanTypesRequired.Signature = true
 					sum.SignatureDigest = info.SignatureDigest
```

Comment:
When newImage is true & imageChanged is true & signatureInfoChanged is true, it reaches here.
sum.SignatureDigest is set to the new signature image digest.
Then in line 1594 the summary is written to kv and scan task is scheduled. 
Does it mean if the scan fails later(because of network, registry, ....), sum.SignatureDigest is already set to the new digest? Is this correct?

Reply:
If the scan fails, then it will set the `SignatureStatus` result in the scan field to `failed` which should then retry the signature scan regardless.